### DISCUSSION TITLE
Hide collection create and edit controls programatically ?

### DISCUSSION THREAD

**Discussion by sksar** [pocketbase/pocketbase#6172] [Q&A] [answered]
Is it possible to check the Hide collection create and edit controls, programatically using Golang code.

Like I am using pocketbase as a framework. And in production I don't want the button to even appear at all. 
No matter what, no superuser can change anything in the production system, the button is turned on by default and cannot be changed from the production admin UI (maybe it can be disabled or hidden in the production if turned on by golang code)

Is that possible ?
If not, then this would a good feature request.

![Screenshot 2024-12-24 at 09 51 20](https://github.com/user-attachments/assets/b70fd104-b249-4635-be09-493dbd639c5f)

**Comment by ganigeorgiev** [comment, marked as answer]
>  And in production I don't want the button to even appear at all.

The UI is not customizable and this is not possible at the moment.

> no superuser can change anything in the production system, the button is turned on by default and cannot be changed from the production admin UI 

If you want to prevent settings change (or to overwrite the superuser submitted data) you can attach to the [`OnSettingsUpdateRequest`](https://pocketbase.io/docs/go-event-hooks/#onsettingsupdaterequest) (e.g. `e.NewSettings.Meta.HideControls = true`). For all other cases you can access the settings via `app.Settings()`.

Godoc at the moment unfortunately doesn't show the embedded fields of unexported fields (see https://github.com/golang/go/issues/6600), aka. the Settings struct fields are not shown in the [package docs](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.23.12/core#Settings) (_I'll consider in the future to export the embedded field or move the unexported fields into the main one_) but for now you can refer to the available struct fields in https://github.com/pocketbase/pocketbase/blob/a8dbca64b2240cf5c6ba1c6b6b5e1495834c2963/core/settings_model.go#L122-L130.
